parity/upgrade.rs

```diff
@@ -105,7 +105,9 @@ fn upgrade_from_version(previous_version: &Version) -> Result<usize, Error> {
 fn with_locked_version<F>(db_path: Option<&str>, script: F) -> Result<usize, Error>
 	where F: Fn(&Version) -> Result<usize, Error>
 {
-	let mut path = db_path.map_or({
+	let mut path = db_path.map_or_else(|| {
```

Comment:
It seems to me like the `db_path` is always `is_some()` (`with_locked_version` is only used in `upgrade`, and `db_path` in `upgrade` is always `Some`). Maybe we can remove `map_or_else` altogether, or am I wrong?

Reply:
I think you're right.